### PR TITLE
[HOPSWORKS-1697] Passwrod recovery for clusters created by Hopsworks-cloud

### PR DESCRIPTION
**Pull request by maismail**
## Make sure there is no duplicate PR for this issue

* **Please check if the PR meets the following requirements**
- [ ] Adds tests for the submitted changes (for bug fixes & features)
- [ ] Passes the tests
- [x] HOPSWORKS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit

* **Post a link to the associated JIRA issue**
https://logicalclocks.atlassian.net/browse/HOPSWORKS-1697

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:
tests: https://github.com/logicalclocks/hops-testing/pull/673

### CODE REVIEW

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/util/VariablesService.java [hunk 1]
```diff
@@ -152,5 +152,14 @@ public Response getSecurityQuestions() {
     };
     return Response.ok().entity(questions).build();
   }
+  
+  @GET
+  @Path("isCloud")
+  @Produces(MediaType.APPLICATION_JSON)
```
Comment (tkakantousis) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/util/VariablesService.java:158]: Why is a new endpoint needed? There are similar endpoints there but they shouldn't be anyway. 
I'd suggest we go with using the {id} in the back-end. For the front-end it's ok to use this https://github.com/logicalclocks/hopsworks/pull/519/files#diff-3104bfea9fcc3f42fafed8cc4326a89bR64

Comment (maismail) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/util/VariablesService.java:158]: ok, that should be fine, i was mainly trying not to use the variable name, also didn't want to replicate the logic that checks if the variable value is empty hence it is cloud

Comment (maismail) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/util/VariablesService.java:158]: @tkakantousis {id} is not using the cached values, it is always making a request to the database

Comment (tkakantousis) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/util/VariablesService.java:158]: The expected load from this endpoint is 1 request per week, so NDB can handle it :D
We don't do rate-limiting on the REST API yet, so a DoS attack could equally happen on any other endpoint

Comment (maismail) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/util/VariablesService.java:158]: Ok, will remove it for now

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/proxies/client/HttpClient.java [hunk 2]
```diff
@@ -116,4 +117,8 @@ public void setAuthorizationHeader(HttpRequest httpRequest) {
   public <T> T execute(HttpHost host, HttpRequest request, ResponseHandler<T> handler) throws IOException {
     return client.execute(host, request, handler);
   }
+  
+  public <T> T execute(HttpUriRequest request, ResponseHandler<T> handler) throws IOException {
```
Comment (kouzant) [inline hopsworks-common/src/main/java/io/hops/hopsworks/common/proxies/client/HttpClient.java:121]: Why do we need this method? Can't we use execute(HttpHost host, HttpRequest request, ResponseHandler<T> handler) ? I think we're doing the same here https://github.com/logicalclocks/hopsworks/blob/master/hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/jupyter/config/JupyterProcessMgr.java#L271

**Pull request merged**